lib/temporal/workflow/state_manager.rb

```diff
@@ -19,8 +19,9 @@ class UnsupportedMarkerType < Temporal::InternalError; end
 
       attr_reader :commands, :local_time
 
-      def initialize(dispatcher)
+      def initialize(dispatcher, task_metadata)
```

Comment:
Why do we need the `task_metadata` here? I'm not sure if it's being used

Reply:
This may be an artifact of the merging of the two refactors around metadata generation. I'll clean it up.